packages/library/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -110,19 +125,34 @@ public class RealmConfiguration private constructor(
      */
     // This constructor is never used at runtime, all calls to it are being rewired by the Realm Compiler Plugin to call
     // the internal secondary constructor with all schema classes mapped to their RealmCompanion.
-    public constructor(path: String? = null, name: String = Realm.DEFAULT_FILE_NAME, schema: Set<KClass<out RealmObject>>) :
-        this(path, name, mapOf())
+    constructor(
```

Comment:
IMO, we shouldn't extend this constructor. This one is just used for the simple case of setting schema/name.

Any other advanced use cases should use the builder pattern. This also removes the need to modify the compiler plugin.